test/http_test.c

```diff
@@ -151,6 +151,72 @@ static int test_http_x509(int do_get)
     return res;
 }
 
+static int test_http_url_ok(const char *url, const char *exp_host, int exp_ssl)
+{
```

Comment:
Thanks for adding some tests

Reply:
Yeah, those were really missing, and having added them gives a good direct feedback on the correctness
using `make test TESTS=test_http`
(while `make test TESTS="test_cmp_cli" OPENSSL_CMP_ASPECTS=connection` 
already contained some tests for the URL parsing).